sorts/cycle_sort.py

```diff
@@ -1,52 +1,53 @@
-# Code contributed by Honey Sharma
-def cycle_sort(array):
-    ans = 0
+"""
+Code contributed by Honey Sharma
+Source: https://en.wikipedia.org/wiki/Cycle_sort
+"""
 
-    # Pass through the array to find cycles to rotate.
-    for cycleStart in range(0, len(array) - 1):
-        item = array[cycleStart]
 
-        # finding the position for putting the item.
-        pos = cycleStart
-        for i in range(cycleStart + 1, len(array)):
+def cycle_sort(array: list) -> list:
+    """
+    >>> cycle_sort([4, 3, 2, 1])
+    [1, 2, 3, 4]
+
+    >>> cycle_sort([-4, 20, 0, -50, 100, -1])
+    [-50, -4, -1, 0, 20, 100]
+
+    >>> cycle_sort([-.1, -.2, 1.3, -.8])
+    [-0.8, -0.2, -0.1, 1.3]
+
+    >>> cycle_sort([])
+    []
+    """
```

Comment:
Can we please break this into two PRs?  The first PR should add the tests (lines 8 thru 20) and once that lands, the second PR could refactor the code.